### PR TITLE
Support tags with namespaces and dashes

### PR DESCRIPTION
**Pull request by dmay**

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -53,9 +53,16 @@ function insertAutoCloseTag(event: vscode.TextDocumentChangeEvent): void {
         if (last2chars === "</") {
             let closeTag = getCloseTag(text, excludedTags);
             if (closeTag) {
+                let nextChar = getNextChar(editor, originalPosition);
+                if(nextChar === ">")
+                    closeTag = closeTag.substr(0, closeTag.length - 1);
```
Comment (formulahendry) [inline src/extension.ts]: Add {}

File: src/extension.ts [hunk 2]
```diff
@@ -53,9 +53,16 @@ function insertAutoCloseTag(event: vscode.TextDocumentChangeEvent): void {
...
                 editor.edit((editBuilder) => {
                     editBuilder.insert(originalPosition, closeTag);
                 });
+                if(nextChar === ">")
+                    // editor.selection points to the / symbol
+                    // We count / + tag + >  =  closeTag.length+2
+                    editor.selection = moveSelectionRight(editor.selection, closeTag.length +2);
```
Comment (formulahendry) [inline src/extension.ts]: You may change selection using Promise like https://github.com/formulahendry/vscode-auto-close-tag/blob/master/src/extension.ts#L73. And you may also use `(originalPosition, closeTag.length +1)`

File: src/extension.ts [hunk 3]
```diff
@@ -104,8 +111,14 @@ function insertCloseTag(): void {
     }
 }
 
+function getNextChar(editor:vscode.TextEditor, position: vscode.Position):string{
```
Comment (formulahendry) [inline src/extension.ts]: space surround `string`

File: src/extension.ts [hunk 4]
```diff
@@ -136,6 +149,12 @@ function getCloseTag(text: string, excludedTags: string[]): string {
     }
 }
 
+function moveSelectionRight(selection: vscode.Selection, shift:number):vscode.Selection{
```
Comment (formulahendry) [inline src/extension.ts]: same here. space surround `vscode.Selection`

File: src/extension.ts [hunk 5]
```diff
@@ -104,8 +111,14 @@ function insertCloseTag(): void {
     }
 }
 
+function getNextChar(editor:vscode.TextEditor, position: vscode.Position):string{
+    let next_position = position.translate(0, 1);
```
Comment (formulahendry) [inline src/extension.ts]: Please use the nextPosition (camel naming) to align with the code style. 😃

Comment (dmay) [inline src/extension.ts]: Habits are pain 😩

**Comment by dmay** [comment]
This is addictive...

**Comment by dmay** [comment]
I have a lot, A LOT of bad XML... 😭

**Comment by formulahendry** [comment]
Thanks @dmay for your contribution! :shipit:

**Pull request merged**